elastio-aws-snapshot-action-tag-integration/elastio-aws-snapshot-action-tag-integration-template.yaml

```diff
@@ -48,21 +48,37 @@ Resources:
         ZipFile: |
           import boto3
           import os
-
+          
           def lambda_handler(event, context):
-              response_elements = event.get('detail', {}).get('responseElements', {})
-              resource_id = response_elements.get('snapshotId')
+              print("event:",event)
+              
+              tag_key_to_monitor = os.environ.get('TagKeyToMonitor')
+              elastio_action_tag_value = os.environ.get('ElastioActionTagValue')
+              tag_keys_to_monitor = [tag.strip() for tag in tag_key_to_monitor.split(',')]
               
               ec2_client = boto3.client('ec2')
-
-              snapshot_tags = ec2_client.describe_tags(Filters=[{'Name': 'resource-id', 'Values': [resource_id]}])['Tags']
               
-              tag_key_to_monitor = os.environ.get('TagKeyToMonitor') 
-              elastio_action_tag_value = os.environ.get('ElastioActionTagValue') 
+              detail = event['detail']
               
-              tag_keys_to_monitor = [tag.strip() for tag in tag_key_to_monitor.split(',')]
+              print("detail:",detail)
+              print("tag_keys_to_monitor:",tag_keys_to_monitor)
               
-              if any(tag['Key'] in tag_keys_to_monitor for tag in snapshot_tags):
+              try:
+                  resource_id = detail['responseElements']['snapshotId']
+              except:
+                  print("snapshotId is null")
+          
+              try:
+                  resource_id = detail['responseElements']['imageId']
+              except:
+                  print("ImageId is null")
```

Comment:
It's odd that a `null` field here would raise an exception.  Are you sure this is necessary?  There ought to be a way to check for the absence of this field w/o triggering an exception.  Note that in Python the `null` value is spelled `Nil`.

Anyway if this works then it's not a big deal, this is more of an FYI.